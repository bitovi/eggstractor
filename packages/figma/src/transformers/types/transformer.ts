import { TokenCollection, TransformerResult } from '../../types';

export type Transformer = (
  tokens: TokenCollection,
  useCombinatorialParsing: boolean,
) => TransformerResult;
