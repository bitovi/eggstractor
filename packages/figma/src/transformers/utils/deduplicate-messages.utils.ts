export function deduplicateMessages(tokens: { warnings?: string[]; errors?: string[] }[]): {
  warnings: string[];
  errors: string[];
} {
  const warningsSet = new Set<string>();
  const errorsSet = new Set<string>();

  tokens.forEach((token) => {
    if (token.warnings) {
      token.warnings.forEach((warning) => warningsSet.add(warning));
    }
    if (token.errors) {
      token.errors.forEach((error) => errorsSet.add(error));
    }
  });

  return {
    warnings: Array.from(warningsSet),
    errors: Array.from(errorsSet),
  };
}
