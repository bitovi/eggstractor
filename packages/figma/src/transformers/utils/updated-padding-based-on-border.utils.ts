const BORDER_PROPERTIES = [
  'border',
  'border-left',
  'border-right',
  'border-top',
  'border-bottom',
  'border-width',
  'border-left-width',
  'border-right-width',
  'border-top-width',
  'border-bottom-width',
] as const;

const PADDING_PROPERTIES = [
  'padding',
  'padding-left',
  'padding-right',
  'padding-top',
  'padding-bottom',
] as const;

type CssUnit = 'px' | 'rem' | 'scss' | 'unknown' | 'none';

type ParsedCssValue =
  | {
      format: 'px' | 'rem' | 'none';
      value: number;
      text: string;
    }
  | {
      value: null;
      format: Exclude<CssUnit, 'px' | 'rem' | 'none'>;
      text: string;
    };

function parseCssValue(text: string): ParsedCssValue {
  // special case: "0" (no unit needed in CSS)
  if (text === '0' || text === '0px' || text === '0rem') {
    return { value: 0, text: '0', format: 'none' };
  }

  if (text.startsWith('$')) {
    return { value: null, text: `#{${text}}`, format: 'scss' };
  }

  // match numbers with px/rem
  const match = /^(-?\d*\.?\d+)(px|rem)?$/.exec(text);

  if (match) {
    const [, num, unit] = match;
    return {
      value: parseFloat(num),
      text,
      format: unit as 'px' | 'rem',
    };
  }

  // fallback
  return { value: null, text, format: 'unknown' };
}

/**
 * Subtracts border value from padding value, returning a valid CSS string.
 * Padding can never be negative, so the result is clamped at 0.
 */
function subtract(padding: string, border: string): string {
  const a = parseCssValue(padding);
  const b = parseCssValue(border);

  // first is zero → result is always "0"
  // second is zero → return the original first string as-is
  if (a.value === 0 || b.value === 0) {
    return a.text;
  }

  // same known units → numeric subtraction, clamped at 0
  if (
    a.format === b.format &&
    (a.format === 'px' || a.format === 'rem') &&
    (b.format === 'px' || b.format === 'rem')
  ) {
    const result = Math.max(0, a.value - b.value);
    return result === 0 ? '0' : `${result}${a.format}`;
  }

  // mismatched/unknown units → calc with clamp (no trimming)
  return `calc(${a.text} - ${b.text})`;
}

interface Padding<T = string> {
  'padding-top': T;
  'padding-right': T;
  'padding-bottom': T;
  'padding-left': T;
}

interface ParsedShorthand {
  top: string;
  right: string;
  bottom: string;
  left: string;
}

const parseShorthand = (value: string): ParsedShorthand => {
  const parts = value.trim().split(/\s+/);

  switch (parts.length) {
    case 1:
      return {
        top: parts[0],
        right: parts[0],
        bottom: parts[0],
        left: parts[0],
      };
    case 2:
      return {
        top: parts[0],
        right: parts[1],
        bottom: parts[0],
        left: parts[1],
      };
    case 3:
      return {
        top: parts[0],
        right: parts[1],
        bottom: parts[2],
        left: parts[1],
      };
    case 4:
      return {
        top: parts[0],
        right: parts[1],
        bottom: parts[2],
        left: parts[3],
      };
    default:
      throw new Error('Invalid shorthand');
  }
};

function toPaddingShorthand(paddings: Partial<Padding>): string {
  // Default missing sides to "0"
  const top = paddings['padding-top'] ?? '0';
  const right = paddings['padding-right'] ?? '0';
  const bottom = paddings['padding-bottom'] ?? '0';
  const left = paddings['padding-left'] ?? '0';

  // Case 1: all equal
  if (top === right && right === bottom && bottom === left) {
    return top;
  }

  // Case 2: top/bottom same, left/right same
  if (top === bottom && right === left) {
    return `${top} ${right}`;
  }

  // Case 3: left/right same
  if (right === left) {
    return `${top} ${right} ${bottom}`;
  }

  // Case 4: all different
  return `${top} ${right} ${bottom} ${left}`;
}

const hasBorderAndPaddingStyles = <T extends { styles: Record<string, string> }>(
  instance: T,
): boolean => {
  const hasPadding = PADDING_PROPERTIES.some((prop) => prop in instance.styles);

  if (!hasPadding) {
    return false;
  }

  return BORDER_PROPERTIES.some((prop) => prop in instance.styles);
};

const parseBorder = (borderStyleValue: string): string => {
  if (borderStyleValue === 'none') return '0';
  return borderStyleValue.split(' ')[0];
};

export const updatePaddingStylesBasedOnBorder = <
  T extends {
    styles: {
      [K in (typeof BORDER_PROPERTIES)[number] | (typeof PADDING_PROPERTIES)[number]]?: string;
    };
  },
>(
  instance: T,
): T => {
  if (!hasBorderAndPaddingStyles(instance)) {
    return instance;
  }

  const borderStyle = instance.styles['border'] ? parseBorder(instance.styles['border']) : null;
  const borderTopStyle = instance.styles['border-top']
    ? parseBorder(instance.styles['border-top'])
    : null;
  const borderRightStyle = instance.styles['border-right']
    ? parseBorder(instance.styles['border-right'])
    : null;
  const borderBottomStyle = instance.styles['border-bottom']
    ? parseBorder(instance.styles['border-bottom'])
    : null;
  const borderLeftStyle = instance.styles['border-left']
    ? parseBorder(instance.styles['border-left'])
    : null;

  const borderWidth = instance.styles['border-width']
    ? parseShorthand(instance.styles['border-width'])
    : null;

  const borders = {
    top:
      instance.styles['border-top-width'] || borderWidth?.['top'] || borderTopStyle || borderStyle,
    right:
      instance.styles['border-right-width'] ||
      borderWidth?.['right'] ||
      borderRightStyle ||
      borderStyle,
    bottom:
      instance.styles['border-bottom-width'] ||
      borderWidth?.['bottom'] ||
      borderBottomStyle ||
      borderStyle,
    left:
      instance.styles['border-left-width'] ||
      borderWidth?.['left'] ||
      borderLeftStyle ||
      borderStyle,
  };

  const paddingStyles = instance.styles['padding']
    ? parseShorthand(instance.styles['padding'])
    : null;

  const paddings: Padding<string | null> = {
    'padding-top': instance.styles['padding-top'] || paddingStyles?.['top'] || null,
    'padding-right': instance.styles['padding-right'] || paddingStyles?.['right'] || null,
    'padding-bottom': instance.styles['padding-bottom'] || paddingStyles?.['bottom'] || null,
    'padding-left': instance.styles['padding-left'] || paddingStyles?.['left'] || null,
  };

  // Only add padding properties if they exist
  const newPaddings: { [K in (typeof PADDING_PROPERTIES)[number]]?: string } = {};

  if (paddings['padding-top']) {
    newPaddings['padding-top'] = borders.top
      ? subtract(paddings['padding-top'], borders.top)
      : paddings['padding-top'];
  }

  if (paddings['padding-right']) {
    newPaddings['padding-right'] = borders.right
      ? subtract(paddings['padding-right'], borders.right)
      : paddings['padding-right'];
  }

  if (paddings['padding-bottom']) {
    newPaddings['padding-bottom'] = borders.bottom
      ? subtract(paddings['padding-bottom'], borders.bottom)
      : paddings['padding-bottom'];
  }

  if (paddings['padding-left']) {
    newPaddings['padding-left'] = borders.left
      ? subtract(paddings['padding-left'], borders.left)
      : paddings['padding-left'];
  }

  // If all individual padding properties were used, use padding shorthand as the output
  const newPaddingValues = Object.values(newPaddings);
  const allIndividualValuesUsed =
    newPaddingValues.length === 4 && newPaddingValues.every((value) => !!value);

  // Only add individual or shorthand padding properties if they exist
  const finalizedPaddingStyles: {
    [K in (typeof PADDING_PROPERTIES)[number]]?: string;
  } = {};

  if (allIndividualValuesUsed) {
    finalizedPaddingStyles['padding'] = toPaddingShorthand(newPaddings);
  } else {
    if (instance.styles['padding-top'] && newPaddings['padding-top']) {
      finalizedPaddingStyles['padding-top'] = newPaddings['padding-top'];
    }

    if (instance.styles['padding-right'] && newPaddings['padding-right']) {
      finalizedPaddingStyles['padding-right'] = newPaddings['padding-right'];
    }

    if (instance.styles['padding-bottom'] && newPaddings['padding-bottom']) {
      finalizedPaddingStyles['padding-bottom'] = newPaddings['padding-bottom'];
    }

    if (instance.styles['padding-left'] && newPaddings['padding-left']) {
      finalizedPaddingStyles['padding-left'] = newPaddings['padding-left'];
    }
  }

  // Strip out any existing padding properties to avoid conflicts
  const {
    padding: _p,
    ['padding-top']: _pt,
    ['padding-right']: _pr,
    ['padding-bottom']: _pb,
    ['padding-left']: _pl,
    ...stylesWithoutPadding
  } = instance.styles;

  return {
    ...instance,
    styles: {
      ...stylesWithoutPadding,
      // Merge in the new padding styles
      ...finalizedPaddingStyles,
    },
  };
};
