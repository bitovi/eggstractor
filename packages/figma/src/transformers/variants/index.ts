export * from './convert-to-generator-tokens';
export * from './convert-variant-group-by';
export * from './generate-combinatorial-styles';
