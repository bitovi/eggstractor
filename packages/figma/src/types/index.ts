export * from './gradients';
export * from './processors';
export * from './processor.types';
export * from './tokens';
