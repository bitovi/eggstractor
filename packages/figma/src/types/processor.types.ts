export interface ProcessorResult {
  value: string;
  warnings?: string[];
  errors?: string[];
}
