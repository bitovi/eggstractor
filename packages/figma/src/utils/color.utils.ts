/**
 * Converts RGBA color components to a hex string or rgba() string.
 */
export function rgbaToString(
  r: number,
  g: number,
  b: number,
  a: number,
): string {
  const rHex = Math.round(r * 255)
    .toString(16)
    .padStart(2, '0');
  const gHex = Math.round(g * 255)
    .toString(16)
    .padStart(2, '0');
  const bHex = Math.round(b * 255)
    .toString(16)
    .padStart(2, '0');

  return a === 1
    ? '#' + rHex + gHex + bHex
    : `rgba(${Math.round(r * 255)}, ${Math.round(g * 255)}, ${Math.round(b * 255)}, ${a})`;
}
