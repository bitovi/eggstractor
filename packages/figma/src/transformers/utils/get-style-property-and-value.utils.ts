import { StyleToken } from '../../types';

export const getStylePropertyAndValue = (
  token: StyleToken,
): Record<string, string> => {
  const output: Record<string, string> = {
    [token.property]: token.rawValue!,
  };

  return output;
};
