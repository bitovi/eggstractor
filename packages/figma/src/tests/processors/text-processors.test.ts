import { collectTokens } from '../../services';
import { ProcessedValue } from '../../types';
import { transformToScss } from '../../transformers';
import { fontProcessors } from '../../processors/font.processor';
import { createTestData } from '../../utils';
import testData from '../fixtures/figma-test-data_paragraph.json';
import testDataAlignment from '../fixtures/figma-test-data-alignment.json';
import testDataFontStyle from '../fixtures/figma-test-data_font-style.json';

describe('Text Processors', () => {
  describe('Text Align Processor', () => {
    const processor = fontProcessors.filter(
      (p) => p.property === 'text-align',
    )[0];

    it('returns null for non-text nodes', async () => {
      const node = {
        type: 'RECTANGLE',
      } as SceneNode;

      const result = await processor.process(new Map(), node);
      expect(result).toBeNull();
    });

    it('processes text alignment correctly', async () => {
      const testCases = [
        {
          name: 'center alignment',
          input: {
            type: 'TEXT',
            textAlignHorizontal: 'CENTER',
          } as SceneNode,
          expected: {
            value: 'center',
            rawValue: 'center',
          },
        },
        {
          name: 'right alignment',
          input: {
            type: 'TEXT',
            textAlignHorizontal: 'RIGHT',
          } as SceneNode,
          expected: {
            value: 'right',
            rawValue: 'right',
          },
        },
        {
          name: 'justified alignment',
          input: {
            type: 'TEXT',
            textAlignHorizontal: 'JUSTIFIED',
          } as SceneNode,
          expected: {
            value: 'justify',
            rawValue: 'justify',
          },
        },
      ];

      for (const testCase of testCases) {
        const result = await processor.process(new Map(), testCase.input);
        expect(result).toEqual(testCase.expected as ProcessedValue);
      }
    });

    it('should process text alignment correctly', async () => {
      const { setupTest } = createTestData(testData);
      const testSetup = await setupTest();

      global.figma = testSetup.figma;

      const tokens = await collectTokens(vi.fn());
      const { result: template } = transformToScss(tokens, false);
      expect(template).toMatchSnapshot('text-alignment');
      const { result: combinatorial } = transformToScss(tokens, true);
      expect(combinatorial).toMatchSnapshot('text-alignment');
    });

    it('should process paragraph alignment correctly', async () => {
      const { setupTest } = createTestData(testDataAlignment);
      const testSetup = await setupTest();

      global.figma = testSetup.figma;

      const tokens = await collectTokens(vi.fn());
      const { result: template } = transformToScss(tokens, false);
      expect(template).toMatchSnapshot('paragraph-alignment');
      const { result: combinatorial } = transformToScss(tokens, true);
      expect(combinatorial).toMatchSnapshot('paragraph-alignment');
    });

    it('should process font style correctly', async () => {
      const { setupTest } = createTestData(testDataFontStyle);
      const testSetup = await setupTest();

      global.figma = testSetup.figma;

      const tokens = await collectTokens(vi.fn());
      const { result: template } = transformToScss(tokens, false);
      expect(template).toMatchSnapshot('font-style');
      const { result: combinatorial } = transformToScss(tokens, true);
      expect(combinatorial).toMatchSnapshot('font-style');
    });
  });
});
