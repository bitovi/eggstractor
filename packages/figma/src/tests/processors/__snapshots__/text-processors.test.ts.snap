// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text Processors > Text Align Processor > should process font style correctly > font-style 1`] = `
"// Generated SCSS Variables
$text-primary: #bfc5c6;
$size-16: 1rem;
$font-style-italic: italic;
$font-weight-normal: normal;
$font-weight-medium: 31.25rem;

// Generated SCSS Mixins
@mixin font-style-italics-variable {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: $font-style-italic;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin font-style-italics-custom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin font-weight-string-variable {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: $font-weight-normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin font-weight-string-custom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: 1rem;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin font-weight-numerical-variable {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Roboto;
  font-size: $size-16;
  font-weight: $font-weight-medium;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
"
`;

exports[`Text Processors > Text Align Processor > should process font style correctly > font-style 2`] = `
"// Generated SCSS Variables
$text-primary: #bfc5c6;
$size-16: 1rem;
$font-style-italic: italic;
$font-weight-normal: normal;
$font-weight-medium: 31.25rem;

// Generated SCSS Mixins
@mixin font-style-italics-variable {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: $font-style-italic;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin font-style-italics-custom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin font-weight-string-variable {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: $font-weight-normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin font-weight-string-custom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: 1rem;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin font-weight-numerical-variable {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Roboto;
  font-size: $size-16;
  font-weight: $font-weight-medium;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
"
`;

exports[`Text Processors > Text Align Processor > should process paragraph alignment correctly > paragraph-alignment 1`] = `
"// Generated SCSS Variables
$text-primary: #bfc5c6;
$size-16: 1rem;

// Generated SCSS Mixins
@mixin frame--top {
  padding: 0rem;
  background: #ffffff;
}
@mixin frame--top-paragraph-alignment-top {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin autolayout--top {
  display: flex;
  flex-direction: column;
  gap: 0.625rem;
  padding: 0rem 1.875rem;
  background: #ffffff;
  justify-content: center;
}
@mixin autolayout--top-paragraph-alignment-top {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-alignment-top {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin frame--middle {
  padding: 0rem;
  background: #ffffff;
}
@mixin frame--middle-paragraph-alignment-center {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin autolayout--middle {
  display: flex;
  flex-direction: column;
  gap: 0.625rem;
  padding: 0rem 1.875rem;
  background: #ffffff;
  justify-content: center;
}
@mixin autolayout--middle-paragraph-alignment-center {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-alignment-center {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin frame--bottom {
  padding: 0rem;
  background: #ffffff;
}
@mixin frame--bottom-paragraph-alignment-bottom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin autolayout--bottom {
  display: flex;
  flex-direction: column;
  gap: 0.625rem;
  padding: 0rem 1.875rem;
  background: #ffffff;
  justify-content: center;
}
@mixin autolayout--bottom-paragraph-alignment-bottom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-alignment-bottom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
"
`;

exports[`Text Processors > Text Align Processor > should process paragraph alignment correctly > paragraph-alignment 2`] = `
"// Generated SCSS Variables
$text-primary: #bfc5c6;
$size-16: 1rem;

// Generated SCSS Mixins
@mixin frame--top {
  padding: 0rem;
  background: #ffffff;
}
@mixin frame--top-paragraph-alignment-top {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin autolayout--top {
  display: flex;
  flex-direction: column;
  gap: 0.625rem;
  padding: 0rem 1.875rem;
  background: #ffffff;
  justify-content: center;
}
@mixin autolayout--top-paragraph-alignment-top {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-alignment-top {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin frame--middle {
  padding: 0rem;
  background: #ffffff;
}
@mixin frame--middle-paragraph-alignment-center {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin autolayout--middle {
  display: flex;
  flex-direction: column;
  gap: 0.625rem;
  padding: 0rem 1.875rem;
  background: #ffffff;
  justify-content: center;
}
@mixin autolayout--middle-paragraph-alignment-center {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-alignment-center {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin frame--bottom {
  padding: 0rem;
  background: #ffffff;
}
@mixin frame--bottom-paragraph-alignment-bottom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin autolayout--bottom {
  display: flex;
  flex-direction: column;
  gap: 0.625rem;
  padding: 0rem 1.875rem;
  background: #ffffff;
  justify-content: center;
}
@mixin autolayout--bottom-paragraph-alignment-bottom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-alignment-bottom {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
"
`;

exports[`Text Processors > Text Align Processor > should process text alignment correctly > text-alignment 1`] = `
"// Generated SCSS Variables
$text-primary: #bfc5c6;
$size-16: 1rem;

// Generated SCSS Mixins
@mixin paragraph-justification-justify {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: 1rem;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-justification-right {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: 1rem;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-justification-center {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-justification-left {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
"
`;

exports[`Text Processors > Text Align Processor > should process text alignment correctly > text-alignment 2`] = `
"// Generated SCSS Variables
$text-primary: #bfc5c6;
$size-16: 1rem;

// Generated SCSS Mixins
@mixin paragraph-justification-justify {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: 1rem;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-justification-right {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: 1rem;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-justification-center {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
@mixin paragraph-justification-left {
  display: flex;
  flex-direction: column;
  color: $text-primary;
  font-family: Inter;
  font-size: $size-16;
  font-weight: 400;
  font-style: normal;
  line-height: normal;
  width: 6.25rem;
  height: 1.25rem;
}
"
`;
